crates/uv-trampoline/src/bounce.rs

```diff
@@ -230,10 +242,16 @@ fn read_trampoline_metadata(executable_name: &Path) -> (TrampolineKind, PathBuf)
         parent_dir.join(path)
     };
 
-    // NOTICE: dunce adds 5kb~
-    let path = dunce::canonicalize(path.as_path()).unwrap_or_else(|_| {
-        error_and_exit("Failed to canonicalize script path");
-    });
+    let path = if !path.is_absolute() || matches!(kind, TrampolineKind::Script) {
+        // // NOTICE: dunce adds 5kb~
+        dunce::canonicalize(path.as_path()).unwrap_or_else(|_| {
+            error_and_exit("Failed to canonicalize script path");
+        })
```

Comment:
Do we want to resolve junctions/symlinks with relative paths? Not critical, but since we're already at it, we could revert this to https://github.com/astral-sh/uv/pull/5750/files#diff-969979506be03e89476feade2edebb4689a9c261f325988d3c7efc5e51de26d1L273-L277 . Otherwise, could you leave a TODO?

Reply:
I was trying to make the minimal change here in the context of this PR. I'll leave a TODO with your link